tools/autograd/gen_variable_type.py

```diff
@@ -550,6 +550,8 @@
     # _nested_tensor_size() should never actually be called with requires_grad=True tensor
     "_nested_tensor_size",
     "_nested_tensor_strides",
+    # Functional collectives keep an internal ref through the Work object
+    "all_reduce",
```

Comment:
all_gather also

Reply:
and i guess @wanchaol youll need to do this manually on your PR since reduce_scatter is not landed yet

Reply:
i didn't see the update, did you push?

Reply:
I forgot to type my PK password and push timedout. It's there now.